crates/holochain/src/core/ribosome/guest_callback/validate.rs

```diff
@@ -71,7 +66,12 @@ impl Invocation for ValidateInvocation {
         }
         match self.element.entry().as_option() {
             Some(Entry::Agent(_)) => fns.push("agent".into()),
-            Some(Entry::App(_)) => fns.push("entry".into()),
+            Some(Entry::App(_)) => {
+                fns.push("entry".into());
+                if let Some(EntryDefId::App(entry_def_id)) = self.entry_def_id.clone() {
+                    fns.push(entry_def_id);
+                }
```

Comment:
What if the def ID contains unicode? or a space? Or even CamelCase? That's going to make for some invalid, or at least weird, function names.

Not sure if this is handled by the CallIterator itself, and maybe this is a broader question than just this PR. I don't like where this is going though, where a user might have to deduce what function to call if their EntryDefId gets mapped into the subset of valid function names.

Reply:
this has always been the case, that if we want to derive callback names from some user provided string, that string needs to be function name compatible

is this a real problem? entry def ids are defined in the same codebase as the (optional) callbacks, so unicode and weird casing simply won't match the callback, and why should it?

Reply:
I don't really see this as an issue as the callback (which is optional btw) has to be compiled in the same wasm as the EntryDefId so you can't use a space, you could do camel case but you'd get a compiler warning